src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -402,23 +402,25 @@ class GrowableArrayWithAllocator : public GrowableArrayView<E> {
 
   void push(const E& elem) { append(elem); }
 
-  E at_grow(int i, const E& fill = E()) {
+  template<typename... Args >
+  E& at_grow(int i, const Args&... args) {
     assert(0 <= i, "negative index %d", i);
     if (i >= this->_len) {
       if (i >= this->_capacity) grow(i);
       for (int j = this->_len; j <= i; j++)
-        this->_data[j] = fill;
-      this->_len = i+1;
+        new (&this->_data[j]) E(args...);
```

Comment:
Use global placement new, e.g.  `::new`.  Also below, in `at_put_grow`

Reply:
Style: Missing braces around the for-loop body.  Also below in `at_put_grow`.

Reply:
I believe you also need to destroy the old object before initialise a new one.